### PR TITLE
feat: add theming support

### LINKED ISSUE #9: Dark Mode
Dark Mode support is basically required nowadays, otherwise the dark mode maximalists will rip us apart. A day/night toggle should be good for starters. We can also do day/night/dusk as seen in the Thunderhub UI:

Screenshot 2022-01-24 at 11 37 31

### PR DESCRIPTION
**Pull request by dennisreimann**
Theme basics for light/dark mode.

Unfortunately Bootstrap v5 still does not offer support for implementing this easily, so we will either need to add overrides for our customizations or generate a separate Bootstrap version, like I did for [BTCPay](https://design.btcpayserver.org/resources/css/).

Will think about this some more :)

Closes #9.

## Preview

image

### CODE REVIEW

File: src/components/App.jsx [hunk 1]
```diff
@@ -201,6 +205,9 @@ export default function App() {
           <rb.Nav.Item>
             <a href="https://twitter.com/joinmarket" target="_blank" rel="noreferrer" className="nav-link text-secondary">Twitter</a>
           </rb.Nav.Item>
+          <rb.Nav.Item>
+            <a href="https://twitter.com/joinmarket" className="nav-link text-secondary" onClick={(e) => { e.preventDefault(); setAndPersistTheme(theme === window.JM.THEMES[0] ? window.JM.THEMES[1] : window.JM.THEMES[0]) }}>{theme === window.JM.THEMES[0] ? 'Dark' : 'Light'} theme</a>
```
Comment (dnlggr) [inline src/components/App.jsx:209]:
```suggestion
            <a href="#" className="nav-link text-secondary" onClick={(e) => { e.preventDefault(); setAndPersistTheme(theme === window.JM.THEMES[0] ? window.JM.THEMES[1] : window.JM.THEMES[0]) }}>{theme === window.JM.THEMES[0] ? 'Dark' : 'Light'} theme</a>
```

**Review by dnlggr — COMMENTED** [review]
Cool—looks pretty good already! 🌞 🌑

**Review by dnlggr — APPROVED** [review]
It already looks great tbh! ✨ 

I'll trust your judgement on whether to go with a custom Bootstrap version or stick with manual CSS overrides. If you decide to go for the latter one—here's my approval for that: ✅

**Comment by dennisreimann** [comment]
Rebased and adapted the settings view:

image

There are still minor kinks like the loading indicator not showing properly on dark mode, but I'd say let's bring this in as we are about to change the loading animation anyways.

**Review by dnlggr — APPROVED** [review]
LGTM ✅

**Pull request merged**